File: game/forest/hurt_flying_falling_hang_animation.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from forest.forest_resources import ForestResources
from forest.hurt_flying_falling_hang_talking import HurtFlyingFallingHangTalking
from state import State


class HurtFlyingFallingHangAnimation(State):
    def process_events(self, events):
        if self.get_frame_index_fast() >= len(ForestResources.catapult_hang):
            return HurtFlyingFallingHangTalking
        return None

    def on_enter(self) -> None:
        super().on_enter()
        AudioHelper.play(ForestResources.sfx_hugo_hangstart, self.context.country)

    def render(self, screen):
        screen.blit(Animation.get_frame(ForestResources.catapult_hang, self.get_frame_index_fast()), (0,0))

File: game-web/src/forest/hurtFlyingFallingHangAnimation.ts
/** Port of game/forest/hurt_flying_falling_hang_animation.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { HurtFlyingFallingHangTalking } from './hurtFlyingFallingHangTalking';
import type { PhoneEvents } from '../phoneEvents';

export class HurtFlyingFallingHangAnimation extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    return this.getFrameIndexFast() >= ForestResources.catapultHang.length ? HurtFlyingFallingHangTalking : null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.sfxHugoHangstart);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(ForestResources.catapultHang, this.getFrameIndexFast()), 0, 0);
  }
}
